src/main/java/org/threadly/litesockets/buffers/ReuseableMergedByteBuffers.java

```diff
@@ -0,0 +1,222 @@
+package org.threadly.litesockets.buffers;
+
+import java.nio.BufferUnderflowException;
+import java.nio.ByteBuffer;
+import java.util.ArrayDeque;
+
+import org.threadly.util.ArgumentVerifier;
+
+/**
+ * This class is used to combine multiple ByteBuffers into 1 simplish to use interface.
+ * It provides most of the features of a single ByteBuffer, but with the ability to perform those 
+ * operations spanning many ByteBuffers.
+ * 
+ * The idea here is to keep from having to copy around and merge ByteBuffers as much as possible. 
+ * 
+ * NOTE: This is not threadSafe.  It should only be accessed by 1 thread at a time.
```

Comment:
Probably worth mentioning here that our client threaded design makes this only a concern if you are accessing it across multiple clients, or other custom created thread pools.